latte/latte.TESTGEN.ps1

```diff
@@ -77,111 +73,142 @@ function test_send {
+                Write-Host "$snd is not a valid send method"
+                $isValid = $false
+            }
+        }
+    } else {
+        Write-Host "SendMethod cannot be null"
+        $isValid = $false
+    }
+    return $isValid
+} # validate_config()
+
 #===============================================
 # External Functions - Main Program
 #===============================================
 function test_main {
     Param(
-        [parameter(Mandatory=$false)] [switch] $Detail = $false,
-        [parameter(Mandatory=$false)] [Int]    $Iterations = 1,
-        [parameter(Mandatory=$false)] [ValidateSet('Sampling','Testing')] [string] $Config = "Sampling",
+        [parameter(Mandatory=$false)] [string] $Config = "Default",
         [parameter(Mandatory=$true)]  [string] $DestIp,
         [parameter(Mandatory=$true)]  [string] $SrcIp,
         [parameter(Mandatory=$true)]  [ValidateScript({Test-Path $_ -PathType Container})] [String] $OutDir = "" 
     )
-    input_display
-    
-    [int]    $g_iters      = $Iterations
-    [bool]   $g_detail     = $Detail
-    [string] $g_DestIp     = $DestIp.Trim()
-    [string] $g_SrcIp      = $SrcIp.Trim()
-    [string] $dir          = (Join-Path -Path $OutDir -ChildPath "latte") 
-    [string] $g_log        = "$dir\LATTE.Commands.txt"
-    [string] $g_logSend    = "$dir\LATTE.Commands.Send.txt"
-    [string] $g_logRecv    = "$dir\LATTE.Commands.Recv.txt" 
-    [string] $g_ConfigFile = ".\latte\LATTE.$Config.Config.ps1"
-
-    New-Item -ItemType directory -Path $dir | Out-Null
-    
-    # Optional - Edit spaces in output path for Invoke-Expression compatibility
-    # $dir  = $dir  -replace ' ','` '
-
-    test_latte_generate -OutDir $dir -ConfigFile $g_ConfigFile
+    try {
+        input_display
+        # get config variables
+        $allConfig = Get-Content -Path "$PSScriptRoot\latte.Config.json" | ConvertFrom-Json
+        [Object] $g_Config     = $allConfig.("Latte$Config")
+        if ($null -eq $g_Config) {
+            Write-Host "Latte$Config does not exist in .\latte\latte.Config.json. Please provide a valid config"
+            Throw
+        }
+        if (-Not (validate_config)) {
+            Write-Host "Latte$Config is not a valid config"
+            Throw
+        }
+        [string] $g_DestIp     = $DestIp.Trim()
+        [string] $g_SrcIp      = $SrcIp.Trim()
+        [string] $dir          = (Join-Path -Path $OutDir -ChildPath "latte") 
+        [string] $g_log        = "$dir\LATTE.Commands.txt"
+        [string] $g_logSend    = "$dir\LATTE.Commands.Send.txt"
+        [string] $g_logRecv    = "$dir\LATTE.Commands.Recv.txt"
+
+        New-Item -ItemType directory -Path $dir | Out-Null
+        
+        # Optional - Edit spaces in output path for Invoke-Expression compatibility
+        # $dir  = $dir  -replace ' ','` '
+
+        test_latte_generate -OutDir $dir
+    } catch {
+        Write-Host "Unable to generate LATTE commands"
```

Comment:
Please also log the reason (exception). for ex:
Write-Host "Exception $($_.Exception.Message) in $($MyInvocation.MyCommand.Name)"